src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -341,6 +342,30 @@ Payment::doApply()
 
     AccountID const dstAccountID(ctx_.tx.getAccountID(sfDestination));
     STAmount const dstAmount(ctx_.tx.getFieldAmount(sfAmount));
+
+    // if the transaction is transaction-level delegated, ctx_.permissions
+    // was already cleaned up.
+    if (ctx_.tx.isDelegated() && !ctx_.permissions.empty())
```

Comment:
should prob create a helper function for the logic `ctx_.tx.isDelegated() && !ctx_.permissions.empty()` so that the same code isn't in every transactor